nwb_conversion_tools/tools/nwb_helpers.py

```diff
@@ -59,3 +59,54 @@ def make_nwbfile_from_metadata(metadata: dict):
     if isinstance(nwbfile_kwargs.get("session_start_time", None), str):
         nwbfile_kwargs["session_start_time"] = datetime.fromisoformat(metadata["NWBFile"]["session_start_time"])
     return NWBFile(**nwbfile_kwargs)
+
+
+def add_device_from_metadata(nwbfile: NWBFile, modality: str = "Ecephys", metadata: dict = None):
+    """
+    Add device information from metadata to NWBFile object.
+
+    Will always ensure nwbfile has at least one device, but multiple
+    devices within the metadata list will also be created.
+
+    Parameters
+    ----------
+    nwbfile: NWBFile
+        nwb file to which the new device information is to be added
+    modality: str
+        Type of data recorded by device. Options:
+        - Ecephys (default)
+        - Icephys
+        - Ophys
+        - Behavior
+    metadata: dict
+        Metadata info for constructing the NWBFile (optional).
+        Should be of the format
+            metadata[modality]['Device'] = [
+                {
+                    'name': my_name,
+                    'description': my_description
+                },
+                ...
+            ]
+        Missing keys in an element of metadata['Ecephys']['Device'] will be auto-populated with defaults.
+    """
+    assert isinstance(nwbfile, NWBFile), "'nwbfile' should be of type pynwb.NWBFile"
+    assert modality in [
+        "Ecephys",
+        "Icephys",
+        "Ophys",
+        "Behavior",
+    ], f"Invalid modality {modality} when creating device."
+
+    defaults = dict(name="Device", description=f"{modality} device. Automatically generated.")
```

Comment:
This default device name should match up with what we decide on above.

Reply:
done